### PR TITLE
chore: add tests/sources for manifest object urls

### PR DESCRIPTION
**Pull request by brandonocasey**
Add tests/sources for Manifest object urls

### CODE REVIEW

File: scripts/index-demo-page.js [hunk 1]
```diff
@@ -61,17 +65,49 @@
...
+  dashManifestXhr.send();
+
   // all relevant elements
   var urlButton = document.getElementById('load-url');
   var sources = document.getElementById('load-source');
   var stateEls = {};
 
   var getInputValue = function(el) {
     if (el.type === 'url' || el.type === 'text' || el.nodeName.toLowerCase() === 'textarea') {
+      if (isManifestObjectType(el.value)) {
```
Comment (brandonocasey) [inline scripts/index-demo-page.js:108]: These object urls are too big to save as a query param so we have to ignore them.

File: scripts/index-demo-page.js [hunk 2]
```diff
@@ -446,6 +482,15 @@
       var ext;
       var type = stateEls.type.value;
 
+      // reset type if it's a manifest object's type
+      if (type === 'application/vnd.videojs.vhs+json') {
```
Comment (brandonocasey) [inline scripts/index-demo-page.js:486]: always reset if the type is 'application/vnd.videojs.vhs+json' as we don't really want to use it between sources.

File: scripts/index-demo-page.js [hunk 3]
```diff
@@ -446,6 +482,15 @@
...
+        type = '';
+      }
+
+      if (isManifestObjectType(stateEls.url.value)) {
```
Comment (brandonocasey) [inline scripts/index-demo-page.js:490]: always sat manifest objects to this type.

File: test/playback.test.js [hunk 4]
```diff
@@ -420,12 +422,14 @@ QUnit[testFn]('zero-length id3 segment', function(assert) {
...
...%2CRESOLUTION%3D640x360%2CAUDIO%3D%22bipbop_audio%22%2CSUBTITLES%3D%22subs%22%0D%0Ahttps%3A%2F%2Fd2zihajmogu5jn.cloudfront.net%2Fbipbop-advanced%2Fgear2%2Fprog_index.m3u8%0D%0A%0D%0A%23EXT-X-STREAM-INF%3ABANDWIDTH%3D915905%2CCODECS%3D%22mp4a.40.2%2C%20avc1.4d401f%22%2CRESOLUTION%3D960x540%2CAUDIO%3D%22bipbop_audio%22%2CSUBTITLES%3D%22subs%22%0D%0Ahttps%3A%2F%2Fd2zihajmogu5jn.cloudfront.net%2Fbipbop-advanced%2Fgear3%2Fprog_index.m3u8%0D%0A%0D%0A%23EXT-X-STREAM-INF%3ABANDWIDTH%3D1030138%2CCODECS%3D%22mp4a.40.2%2C%20avc1.4d401f%22%2CRESOLUTION%3D1280x720%2CAUDIO%3D%22bipbop_audio%22%2CSUBTITLES%3D%22subs%22%0D%0Ahttps%3A%2F%2Fd2zihajmogu5jn.cloudfront.net%2Fbipbop-advanced%2Fgear4%2Fprog_index.m3u8%0D%0A%0D%0A%23EXT-X-STREAM-INF%3ABANDWIDTH%3D1924009%2CCODECS%3D%22mp4a.40.2%2C%20avc1.4d401f%22%2CRESOLUTION%3D1920x1080%2CAUDIO%3D%22bipbop_audio%22%2CSUBTITLES%3D%22subs%22%0D%0Ahttps%3A%2F%2Fd2zihajmogu5jn.cloudfront.net%2Fbipbop-advanced%2Fgear5%2Fprog_index.m3u8%0D%0A%0D%0A%23EXT-X-STREAM-INF%3ABANDWIDTH%3D41457%2CCODECS%3D%22mp4a.40.2%22%2CAUDIO%3D%22bipbop_audio%22%2CSUBTITLES%3D%22subs%22%0D%0Ahttps%3A%2F%2Fd2zihajmogu5jn.cloudfront.net%2Fbipbop-advanced%2Fgear0%2Fprog_index.m3u8';
```
Comment (brandonocasey) [inline test/playback.test.js:425]: moved the actual uris out of the tests here so that they can actually be readable.

**Review by gesinger — COMMENTED** [review]
It looks good, but the Netlify preview breaks from the new local XHR requests.

**Review by gesinger — APPROVED** [review]

**Pull request merged**